examples/features/deadline/client/main.go

```diff
@@ -0,0 +1,95 @@
+/*
+ *
+ * Copyright 2018 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Binary client is an example client.
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	pb "google.golang.org/grpc/examples/features/proto/echo"
+	"google.golang.org/grpc/status"
+)
+
+func unaryCall(c pb.EchoClient, requestID int, message string, want codes.Code) {
+	// Creates a context with a one second deadline for the RPC.
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	req := &pb.EchoRequest{Message: message}
+
+	_, err := c.UnaryEcho(ctx, req)
+	got := status.Code(err)
+	fmt.Printf("[%v] wanted = %v, got = %v\n", requestID, want, got)
+}
+
+func streamingCall(c pb.EchoClient, requestID int, message string, want codes.Code) {
+	// Creates a context with a one second deadline for the RPC.
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	stream, err := c.BidirectionalStreamingEcho(ctx)
+	if err != nil {
+		log.Printf("Stream err: %v", err)
+		return
+	}
+
+	err = stream.Send(&pb.EchoRequest{Message: message})
+	if err != nil {
+		log.Printf("Send error: %v", err)
+		return
+	}
+
+	_, err = stream.Recv()
+
+	got := status.Code(err)
+	fmt.Printf("[%v] wanted = %v, got = %v\n", requestID, want, got)
+}
+
+func main() {
+	port := flag.Int("port", 50052, "port number")
```

Comment:
Please see https://github.com/grpc/grpc-go/blob/master/examples/features/metadata/client/main.go#L35 -- we usually declare all flags as package globals since they apply to the whole binary.